model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/schema.proto

```diff
@@ -110,6 +112,27 @@ message LogicalType {
   FieldValue argument = 5;
 }
 
+// Universally defined Logical types for Row schemas.
+// These logical types are supposed to be understood by all SDKs.
+message LogicalTypes {
+  enum Enum {
+    // A URN for Python Callable logical type
+    //   - Representation type: STRING
+    //   - Language type: In Python SDK, PythonCallableWithSource.
+    //     In any other SDKs, a wrapper object for a string which
+    //     can be evaluated to a Python Callable object.
+    PYTHON_CALLABLE = 0 [(org.apache.beam.model.pipeline.v1.beam_urn) =
+        "beam:logical_type:python_callable:v1"];
+
+    // A URN for MicrosInstant type
+    //   - Representation type: ROW<seconds: INT64, micros: INT64>
+    //   - A timestamp without a timezone where seconds + micros represents the
+    //     amount of time since the epoch.
```

Comment:
There's also "beam:logical_type:schema:v1".

Can you also cleanup (remove these from the comment) and update the usages ?

Reply:
(we can create a Jira if this is a large effort somehow).

Reply:
For now, python_callable and micros_instant are only two logical types implemented in both Java and Python SDKs. schema logical type is implemented in Java SDK but not in Python SDK. There're also a bunch of other logical types only available to Java SDKs. Do you think we should add those to schema.proto too?

Reply:
And what do you mean by "update the usages" ?

Reply:
Probably we should move all logical types that are expected to be understood by all SDKs to here and remove them from the definition in the comment. "beam:logical_type:micros_instant:v1"" and "beam:logical_type:schema:v1" seems to be the only two logical types defined there (even though latter is only supported by Java currently).

By "update the usages", I meant update the Java/Python classes where the URN is defined to refer to the proto message instead. Seems like you already did this for "beam:logical_type:micros_instant:v1".

Reply:
Moving all existing urns seems to be out of scope for this PR. I created a separate Jira ticket for it: https://issues.apache.org/jira/browse/BEAM-14468

Reply:
Thanks. But can you remove the URNs defined here from the beam_runner_api.proto and and may be add a comment there to define new logical types here ?

Reply:
Done.